sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -229,42 +229,183 @@ CURLcode CurlSession::HttpRawSend()
 CURLcode CurlSession::ReadStatusLineAndHeadersFromRawResponse()
 {
   auto parser = ResponseBufferParser();
+  auto bufferSize = uint64_t();
 
   // Keep reading until all headers were read
   while (!parser.IsParseCompleted())
   {
     // Try to fill internal buffer from socket.
     // If response is smaller than buffer, we will get back the size of the response
-    auto bufferSize = ReadSocketToBuffer(this->m_readBuffer, LibcurlReaderSize);
+    bufferSize = ReadSocketToBuffer(this->m_readBuffer, LibcurlReaderSize);
 
     // parse from buffer to create response
     auto bytesParsed = parser.Parse(this->m_readBuffer, (size_t)bufferSize);
     // if end of headers is reach before the end of response, that's where body start
-    if (bytesParsed < bufferSize)
+    if (bytesParsed + 2 < bufferSize)
     {
-      this->m_bodyStartInBuffer = bytesParsed + 1; // Set the start of body (skip \n)
+      this->m_bodyStartInBuffer = bytesParsed + 1; // Set the start of body (skip \r)
     }
   }
 
   this->m_response = parser.GetResponse();
+  this->m_innerBufferSize = static_cast<size_t>(bufferSize);
 
   // For Head request, set the length of body response to 0.
   if (this->m_request.GetMethod() == HttpMethod::Head)
   {
-    this->m_response->SetBodyStream(new CurlBodyStream(0, this));
+    this->m_response->SetBodyStream(std::make_unique<CurlBodyStream>(0, this));
     return CURLE_OK;
   }
 
   // TODO: tolower ContentLength
   auto headers = this->m_response->GetHeaders();
-  auto bodySize = atoi(headers.at("Content-Length").data());
-  // content-length is used later by session and session won't have access to the response any more
-  // (unique_ptr), so we save this value
-  this->m_contentLength = bodySize;
-  // Move session to live inside the stream from response.
-  this->m_response->SetBodyStream(new CurlBodyStream(bodySize, this));
 
-  return CURLE_OK;
+  auto isContentLengthHeaderInResponse = headers.find("Content-Length");
+  if (isContentLengthHeaderInResponse != headers.end())
+  {
+    // Response with Content-Length
+    auto bodySize = atoi(headers.at("Content-Length").data());
```

Comment:
`atoi` cannot handle value > INT_MAX, which is `2GB-1`. use `std::stoull` or `std::stoll` instead